### ISSUE TITLE
avoid using `deployArtifact` from the device-type endpoint

### ISSUE THREAD

**Issue by klutchell** [balena-io-modules/balena-preload#265]
We should find a better way to detect if the image is a flasher type or not, without an API query.

**Comment by pdcastro** [comment]
> without an API query

For what it is worth, specifically with regard to the deprecation of the balena SDK's `models.config.getDeviceTypes()` method and the `/device-types/v1` API endpoint, we could still make use of the SDK / API to obtain device type information, just using different SDK methods / different API endpoints. For example:
* https://github.com/balena-io/balena-cli/pull/2317/files
* https://github.com/balena-io/balena-cli/pull/2324/files

**Comment by thgreasi** [comment]
@klutchell WRT `device_type["yocto"]["deployArtifact"]` in the python script, it looks like it's predictable to some degree and we could potentially search in the FS for what actually exists, right?
The pattern of options that I see is:  `(balena|resin)-image-flasher-(<slug>|<alias>.(balenaos|resinos)-img`

https://github.com/balena-io-modules/balena-preload/blob/f1c791c1c8590f4fd812552fd5088975f653e758/src/preload.py#L716-L717

**Comment by klutchell** [comment]
> Could we potentially search in the FS for whether any of the options actually exists

@thgreasi Yeah, the names are predictable so this is what I thought of as well.
